compiler/codegen.js

```diff
@@ -2086,12 +2086,20 @@ const generateCall = (scope, decl, _global, _name, unusedValue = false) => {
   let args = decl.arguments;
   if (func && args.length < paramCount) {
     // too little args, push undefineds
-    args = args.concat(new Array(paramCount - args.length).fill(DEFAULT_VALUE));
+    if (func.hasRestArgument) {
+      args = args.concat(new Array(paramCount - 1 - args.length).fill(DEFAULT_VALUE));
+    } else {
+      args = args.concat(new Array(paramCount - args.length).fill(DEFAULT_VALUE));
+    }
   }
 
-  if (func && args.length > paramCount) {
-    // too many args, slice extras off
-    args = args.slice(0, paramCount);
+  if (func && func.hasRestArgument) {
```

Comment:
Instead of having two separate checks for the flag could we just have one for both the differences?